src/app/admin/product-form/product-form.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Product } from './../../models/product';
+import { ProductService } from './../../product.service';
+import { CategoryService } from './../../category.service';
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-product-form',
+  templateUrl: './product-form.component.html',
+})
+export class ProductFormComponent implements OnInit {
+  categories$;
```

Comment:
you are using typescript so everything should have a declared type. This is an observable so I would rework this line to say categoriesObservable: Observable<Category>